android/src/main/java/com/rudderstack/android/internal/infrastructure/ReinstatePlugin.kt

```diff
@@ -69,84 +60,83 @@ internal class ReinstatePlugin : InfrastructurePlugin {
     override fun setup(analytics: Analytics) {
         _analytics = analytics
         setReinstated(false)
-    }
-
-    override fun updateRudderServerConfig(config: RudderServerConfig) {
-        val sourceId = config.source?.sourceId ?: return
-        this.sourceId = sourceId
-        reinstate()
-    }
-
-    override fun updateConfiguration(configuration: Configuration) {
         reinstate()
     }
 
     private fun reinstate() {
-        if (_isReinstated.get()) return
```

Comment:
Are we sure we don't need this flag? As we are setting this flag but not utilising it. If it's not needed can we please consider removing this flag altogether?